funds_monitor/src/avail.rs

```diff
@@ -31,135 +28,145 @@ pub async fn run(cfg: Arc<Config>) -> Result<(), ClientError> {
         error(&format!("Failed to sync database: {}", e.to_string()));
     }
 
-    let mut stream = sdk.client.blocks().subscribe_finalized().await?;
-    while let Some(avail_block) = stream.next().await {
-        match avail_block {
-            Ok(avail_block) => {
-                let block: Block = match Block::from_block(avail_block).await {
-                    Ok(block) => block,
-                    Err(e) => {
-                        error(&format!("Error fetching block: {}", e));
-                        continue;
-                    }
-                };
-                info(&format!("Avail block number: {}", block.block.number()));
-
-                if let Err(e) = process_block(block, &utils, &cfg.avail_deposit_address).await {
-                    error(&format!("Failed to sync database: {}", e.to_string()));
-                }
-            }
-            Err(e) => {
-                error(&format!("Error fetching block: {}", e));
+    let mut sub = Sub::new(sdk.clone());
+    loop {
+        let b_info = sub.next().await;
+        let b_info = match b_info {
+            Ok(x) => x,
+            Err(err) => {
```

Comment:
This won't skip a blocks. Once it gets back it will retry to fetch the same block.